crates/apub/src/activities/community/announce.rs

```diff
@@ -185,3 +193,23 @@ impl TryFrom<AnnouncableActivities> for RawAnnouncableActivities {
     serde_json::from_value(serde_json::to_value(value)?)
   }
 }
+
+/// Check if an activity in the given community can be accepted. To return true, the community must
+/// either be local to this instance, or it must have at least one local follower.
+///
+/// TODO: This means mentions dont work if the community has no local followers. Can be fixed
+///       by checking if any local user is in to/cc fields of activity. Anyway this is a minor
+///       problem compared to receiving unsolicited posts.
+async fn can_accept_activity_in_community(
```

Comment:
So this could happen if someone subs, then unsubs, leaving stranded, federated communities getting updates?

Reply:
The problem this PR solves is that currently there is no check if we actually follow a community when receiving an activity. So a malicious instance could send activities for its communities to all other instances, and they would blindly accept and display that content. With this fix, activities will be rejected if the community doesnt have any local followers.

If an unfollow gets properly federated then the community wont send any activities to us so everything is fine. However its possible that the unfollow doesnt go through, then it will keep sending activities which will get rejected.

Reply:
@dessalines FWIW, in an *Exploration Post* I hope to get to write at some point, I will make the recommendation that all outboxed remote communities should receive activity updates, even if no local subscriber exists.

How the remote instance will know to send those activities, and how it can be asked to stop, doesn't exist in lemmy today obviously, and will be a core part of the exploration.